### DISCUSSION TITLE
grblHAL on BTT GTR V1.0 Help

### DISCUSSION THREAD

**Discussion by RJ5X** [grblHAL/core#474] [Q&A]
Hi, I am trying to use grblHAL on "Btt GTR V1.0"  for a 5-axis machine.
The "Btt GTR V1.0" is capable of up to 11-Axis and its processor is the (STM32F407IGT6), so similar to (STM32F407ZGT6) on the "BTT SKR PRO V1.1" board. 
I started modifying "btt_skr_pro_v1_1_map.h" and a few more files using VSCode, at this point I can compile without errors and load the binary to the board, using the SDCard.
To be clear, My programming level is not expertise but I have some basic knowledge. I have done a few projects using ST Microcontrollers on STMCubeIDE and Microchip some time ago. 
Now the problem: When I connect the "Btt GTR V1.0" using the onboard USB to the PC, windows reports "Unknow USB Device".
![Unknown USB](https://github.com/grblHAL/core/assets/165228027/da6fb649-6cf1-4f46-a699-c3fb72400b4c)

Can someone guide me on how to solve this first problem and make this board work?
Attached is my project and  "Btt GTR V1.0" schematic.
[GTR V1.0(SCH).PDF](https://github.com/grblHAL/core/files/14783539/GTR.V1.0.SCH.PDF)
[BTT_GTR-V1.0_STM32F407IGT6.zip](https://github.com/grblHAL/core/files/14783543/BTT_GTR-V1.0_STM32F407IGT6.zip)

Thanks,

**Comment by terjeio** [comment]
Which build configuration are you using? It is selected via the "hammer" tool in the STMCubeIDE toolbar.

**Reply by RJ5X** [reply]
Hi Terjeio,

Thanks for your fast response,

I am compiling on VSCode with plataformIO, not STMCubeIDE. 
I can be in a error, because I am not so familiar with this different building configurations. But I think that I set VSCode to compile my new "BTT GTR V1.0". That build configuration was created from a modified "BTT SKR PRO V1.1" build configuration.

**Reply by terjeio** [reply]
It could be that you have been hit by a bug that was recently fixed - it caused a hardfault if the PWM spindle fails to initialize.
This since I see that you have specified a port/pin combo (PG5) for the PWM output that is not supported - neither [by the driver](https://github.com/grblHAL/STM32F4xx/blob/17bcd213c359a1ee5b49466192a4e0004d6a11b6/Src/pwm.c#L34-L97) nor by the MCU.

Further ports/pins for PWM output can be added if one of the supported ones cannot be used - if PWM capable and not bound to a timer used for a different purpose.

As a first step update to the latest version.

**Reply by RJ5X** [reply]
Hi Terjeio,

I will do it tonight.


Thanks,

**Comment by RJ5X** [comment]
Hi Terjeio,

After updating with the latest file, windows recognize it as "STMicroelectronics Virtual COM Port".
![STM Virtual COM ](https://github.com/grblHAL/core/assets/165228027/c77d270e-1a8d-425b-95aa-590c1c70681b)
It's good progress. Thanks for it.

Now, I have a few more questions.
1-  I found that "driver.json" and "platafomio.ini" have some sections dedicated to the board characteristics. Do I have to prepare one for my board, since it has a different microcontroller?
![driver json](https://github.com/grblHAL/core/assets/165228027/1fa589c6-5137-45c2-9c76-9d0670acfc1a)
![platoformio ini](https://github.com/grblHAL/core/assets/165228027/6e9ae3ea-0f9b-478b-baf8-60bbb1b09b8c)

2-  About this "xx_FLASH.ld files" LinkerScript.ld. Do I require one like that for my microcontroller (STM32F407IGT6)?
![xx_FLASH ld](https://github.com/grblHAL/core/assets/165228027/8f88e637-e518-4b91-845e-67e258269983)

Thanks a lot for your help,

**Reply by terjeio** [reply]
_driver.json_ is only used by the [Web Builder](http://svn.io-engineering.com:8080) so not strictly neccesary to add the board, but helpful if other users want to build the firmware without installing a toolchain. A bit of the same for _platformio.ini_ - helpful for users wanting to build using that toolchain.

There is no need to add a separate .ld file since the IGT6 variant is covered by the 407VGTX (same amount of RAM and flash).
